e2e/config/config.go

```diff
@@ -6,24 +6,31 @@ import (
 )
 
 var (
-	BuildID           = getEnvWithDefaultIfEmpty(os.Getenv("BUILD_ID"), "local")
-	SubscriptionID    = getEnvWithDefaultIfEmpty("SUBSCRIPTION_ID", "8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8")
-	Location          = getEnvWithDefaultIfEmpty("LOCATION", "eastus")
-	ResourceGroupName = "abe2e-" + Location
-	TagsToRun         = os.Getenv("TAGS_TO_RUN")
-	TagsToSkip        = os.Getenv("TAGS_TO_SKIP")
-	KeepVMSS          = strings.EqualFold(os.Getenv("KEEP_VMSS"), "true")
-	Azure             = MustNewAzureClient(SubscriptionID)
-	// ADO tags every SIG image version with `branch` tag. By specifying `branch=refs/heads/master` we load latest image version from the master branch.
-	SIGVersionTagName             = getEnvWithDefaultIfEmpty("SIG_VERSION_TAG_NAME", "branch")
-	SIGVersionTagValue            = getEnvWithDefaultIfEmpty("SIG_VERSION_TAG_VALUE", "refs/heads/master")
-	IgnoreScenariosWithMissingVHD = strings.EqualFold(os.Getenv("IGNORE_SCENARIOS_WITH_MISSING_VHD"), "true")
+	Azure                         = MustNewAzureClient(SubscriptionID)
+	BuildID                       = lookupEnvWithDefaultString(os.Getenv("BUILD_ID"), "local")
+	IgnoreScenariosWithMissingVHD = lookupEnvWithDefaultBool("IGNORE_SCENARIOS_WITH_MISSING_VHD", false)
+	KeepVMSS                      = lookupEnvWithDefaultBool("KEEP_VMSS", false)
+	Location                      = lookupEnvWithDefaultString("LOCATION", "eastus")
+	ResourceGroupName             = "abe2e-" + Location
+	SIGVersionTagName             = lookupEnvWithDefaultString("SIG_VERSION_TAG_NAME", "branch") // ADO tags every SIG image version with `branch` tag. By specifying `branch=refs/heads/master` we load latest image version from the master branch.
+	SIGVersionTagValue            = lookupEnvWithDefaultString("SIG_VERSION_TAG_VALUE", "refs/heads/master")
+	SkipTestsWithSKUCapacityIssue = lookupEnvWithDefaultBool("SKIP_TESTS_WITH_SKU_CAPACITY_ISSUE", false)
+	SubscriptionID                = lookupEnvWithDefaultString("SUBSCRIPTION_ID", "8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8")
+	TagsToRun                     = os.Getenv("TAGS_TO_RUN")
+	TagsToSkip                    = os.Getenv("TAGS_TO_SKIP")
 )
 
-func getEnvWithDefaultIfEmpty(env string, defaultValue string) string {
+func lookupEnvWithDefaultString(env string, defaultValue string) string {
 	val, ok := os.LookupEnv(env)
 	if !ok {
 		return defaultValue
 	}
 	return val
 }
+
+func lookupEnvWithDefaultBool(env string, defaultValue bool) bool {
```

Comment:
don't have to do it now, but would be nice if we could use generics here instead

Reply:
I find generics are hard to read. Maybe when we have more than 3 types.

Reply:
I don't think generics are going to work. We perform a string to a type conversion. Which is different for every type.